lib/Parser.vala

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2018 elementary LLC. (https://github.com/elementary/Vala-Lint)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ */
+
+public class ValaLint.Parser : Object {
+
+    // For ParseDetailType
+    string[] start_patterns = {
+        "(\\/\\/)",
```

Comment:
Can I suggest you use verbatim strings for all of your regular expressions. i.e. `"""regex here"""`.

https://wiki.gnome.org/Projects/Vala/Tutorial#Strings

You won't need anywhere near as many backslashes for escaping and it will make them much more readable.

Reply:
Sure, thanks!

Reply:
Sure, done!